frontend/src/components/CreateTemplateCard/CreateTemplateCard.tsx

```diff
@@ -1,15 +1,19 @@
 import WorkoutTemplateModal from "../WorkoutTemplateModal/WorkoutTemplateModal";
 import styles from "./CreateTemplateCard.module.css";
-import { useState } from "react";
+import React, { useState } from "react";
 
-const CreateTemplateCard = () => {
+type CreateTemplateCardProps = {
+  fetchTemplates: () => void;
+};
+
+const CreateTemplateCard: React.FC<CreateTemplateCardProps> = ({ fetchTemplates }) => {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const handleClick = () => {
     setIsModalOpen(true);
   };
 
   const closeModal = () => {
-    console.log("Closing modal", "open? ", isModalOpen);
+    fetchTemplates();
```

Comment:
nit: is it possible to only re-fetch templates when a user actually creates a new template (eg. on the success handler for the post request)?

Reply:
Yeah, thanks for the suggestion. I just added this in a new PR